src/main/java/echoserver/EchoServer.java

```diff
@@ -3,10 +3,23 @@
 import java.io.IOException;
 
 public class EchoServer {
-    public void echo(Connection connection) throws IOException {
-        String input;
-        while ((input = connection.read()) != null) {
-            connection.write(input);
+    private final Listenable listener;
+    private final Client echoClient;
+
+    public EchoServer(Listenable listener, Client echoClient) {
+        this.listener = listener;
+        this.echoClient = echoClient;
+    }
+
+    public void connect() throws IOException {
```

Comment:
`connect` doesn't feel quite accurate to me. I think of connect as happening per client, but this is called when we start the server.

Reply:
Okay will change